pkg/graveler/committed/apply.go

```diff
@@ -267,6 +279,15 @@ func (a *applier) applyDiffRangeSourceKey(diffRange *Range, sourceValue *gravele
 
 func (a *applier) applyBothRanges(diffRange *Range, sourceRange *Range) error {
 	switch {
+	case !diffRange.Tombstone && bytes.Equal(sourceRange.MinKey, diffRange.MinKey) && bytes.Equal(sourceRange.MaxKey, diffRange.MaxKey):
+		// insert diff move both
+		if err := a.writer.WriteRange(*diffRange); err != nil {
```

Comment:
How can you skip this range?
The ranges have the same min-max keys, but the `diffRange` only contains _changes_ to be applied on the source range. You can't write it entirely, if I understand correctly.

Reply:
Synced f2f: there is an assumption here that if both ranges here have the same boundaries - then `diffRange` can be written to the result as is, and there is no need to look at `sourceRange`.

This is a strange assumption, and I see it as a bug in the apply. Will think of a way to change the contract of the applyer to make it seem more reasonable, and will update here.

Reply:
The same bounds case should be similar to the same record case. They both state that X was changed and we need to insert only the change ( doesn't matter if we are referencing a record or a range )
I set the case of overlapping bounds as an error for now, hope it will make it more clear.